param/__init__.py

```diff
@@ -2912,18 +2998,24 @@ def _validate_value(self, val, allow_None):
             return
 
         if not isinstance(val, tuple):
-            raise ValueError("DateRange parameter {!r} only takes a tuple value, "
-                             "not {}.".format(self.name, type(val).__name__))
+            raise ValueError(
+                f"{_validate_error_prefix(self)} only takes a tuple value, "
+                f"not {type(val).__name__!r}."
+            )
         for n in val:
             if isinstance(n, dt_types):
                 continue
-            raise ValueError("DateRange parameter {!r} only takes date/datetime "
-                             "values, not type {}.".format(self.name, type(n).__name__))
+            raise ValueError(
+                f"{_validate_error_prefix(self)} only takes date/datetime "
+                f"values, not type {type(n).__name__!r}."
```

Comment:
```suggestion
                f"values, not {type(n)}."
```

Reply:
That version is fine too, thanks! I believe there are error message checks in the test that will need to be updated to match.